### PR TITLE
Bug 906176: Move to using Django staticfiles for media.

### PR DESCRIPTION
**Pull request by pmclanahan**
This keeps the URL for CSS/JS/Images/Etc. as /media/ so that
all of the .less files and urls around the net need not change.
The new MEDIA_URL is '/user_media/', and the STATIC_URL is '/media/'.
The files all stay in the 'media' folder in the project, but they'll now
be served from the 'static' folder after 'collectstatic' is run.

All templates, views, and tests have been updated for the new changes,
as well as the l10n media helpers. The site works well for me in
spot-checks, and the tests pass.

One caveat is that you must run 'manage.py collectstatic' before
running tests due to the way jingo-minify works regarding the
compilation of .less files when not in DEBUG mode. I've yet to find
a way around this.

### CODE REVIEW

File: bedrock/settings/__init__.py [hunk 1]
```diff
@@ -45,5 +45,16 @@
         'CULL_FREQUENCY': 4,  # 1/4 entries deleted if max reached
     }
 }
+# cache for static files
+CACHES['staticfiles'] = {
```
Comment (pmclanahan) [inline bedrock/settings/__init__.py]: Just doing this until we can move to Django 1.7 or add Whitenoise and use the manifest cache.

File: bedrock/settings/base.py [hunk 2]
```diff
@@ -106,10 +106,15 @@ def JINJA_CONFIG():
         'auto_reload': True,
     }
 
-STATIC_URL = '/static/'
+MEDIA_URL = '/user-media/'
```
Comment (pmclanahan) [inline bedrock/settings/base.py]: This is just for now until we get all the s3 jazz working.

**Comment by pmclanahan** [comment]
@jgmize just added another commit that enables the caching staticfiles storage and removes all of the `whatnot.png?2015-01` junk from the site. I believe we can safely rely on staticfiles for all of this.

**Comment by pmclanahan** [comment]
Had to do this because the `static()` helper was munging the urls that included the `?` and not all of them could be moved outside of the helper (e.g. the `high_rez_image()` ones).

**Comment by pmclanahan** [comment]
@jgmize I'm thinking we don't need to block this on the RNA fix right? We don't (shouldn't) currently use the RNA admin on bedrock right?

**Comment by jgmize** [comment]
@pmclanahan agreed

**Comment by pmclanahan** [comment]
WOO \o/ It passed on both 2.6 and 2.7. Need more testing on 2.6, but may be okay. I know whitenoise isn't tested on 2.6, but the bit we're using is okay for now.

**Comment by alexgibson** [comment]
@pmclanahan - just wondering, is demo4 up to date with the changes here? I just tried opening a few of the tour URL's and they don't seem to be loading the relevant JS bundles. e.g.

https://www-demo4.allizom.org/en-US/firefox/35.0/firstrun/
https://www-demo4.allizom.org/en-US/firefox/33.1/whatsnew/?oldversion=28.0
https://www-demo4.allizom.org/en-US/firefox/34.0/whatsnew/?oldversion=28.0
https://www-demo4.allizom.org/en-US/firefox/35.0/whatsnew/?oldversion=29.0

When I inspect the page, all I see is the regular `firefox-bundle` and `tabzilla` JS

**Comment by pmclanahan** [comment]
Looking at the source, it appears the tabzilla.js script tag got hosed up somehow. Looks like a missing double quote at the end of the src attribute. Looking into it. Good catch.

**Comment by pmclanahan** [comment]
Yup. I'd accidentally a `"` from the `base.html` template. So all non-responsive pages would have had this problem. Fix deploying to demo4 now.

**Comment by alexgibson** [comment]
Heads up there's two new 'media' occurrences that just merged in #2704, which will likely need updating here.

**Comment by pmclanahan** [comment]
Thanks Alex!

**Pull request merged**